packages/arb-token-bridge-ui/src/state/cctpState.ts

```diff
@@ -0,0 +1,638 @@
+import dayjs from 'dayjs'
+import {
+  ChainDomain,
+  CompletedCCTPTransfer,
+  PendingCCTPTransfer,
+  Response
+} from '../pages/api/cctp/[type]'
+import { CommonAddress } from '../util/CommonAddressUtils'
+import { shouldTrackAnalytics, trackEvent } from '../util/AnalyticsUtils'
+import { useAccountType } from '../hooks/useAccountType'
+import { useNetworksAndSigners } from '../hooks/useNetworksAndSigners'
+
+// see https://developers.circle.com/stablecoin/docs/cctp-technical-reference#block-confirmations-for-attestations
+export function getBlockBeforeConfirmation(
+  chainId: CCTPSupportedChainId | undefined
+) {
+  if (!chainId) {
+    return 65
+  }
+
+  return {
+    [ChainId.Mainnet]: 65,
+    [ChainId.ArbitrumOne]: 65,
+    [ChainId.Goerli]: 5,
+    [ChainId.ArbitrumGoerli]: 5
+  }[chainId]
+}
+
+export type CCTPSupportedChainId =
+  | ChainId.Mainnet
+  | ChainId.Goerli
+  | ChainId.ArbitrumOne
+  | ChainId.ArbitrumGoerli
+
+function getSourceChainIdFromSourceDomain(
+  sourceDomain: ChainDomain,
+  chainId: ChainId
+): CCTPSupportedChainId {
+  const { isTestnet } = isNetwork(chainId)
+
+  // Deposits
+  if (sourceDomain === ChainDomain.Mainnet) {
+    return isTestnet ? ChainId.Goerli : ChainId.Mainnet
+  }
+
+  // Withdrawals
+  return isTestnet ? ChainId.ArbitrumGoerli : ChainId.ArbitrumOne
+}
+
+export function getUsdcTokenAddressFromSourceChainId(
+  sourceChainId: CCTPSupportedChainId
+) {
+  return {
+    [ChainId.Mainnet]: CommonAddress.Mainnet.USDC,
+    [ChainId.ArbitrumOne]: CommonAddress.ArbitrumOne.USDC,
+    [ChainId.Goerli]: CommonAddress.Goerli.USDC,
+    [ChainId.ArbitrumGoerli]: CommonAddress.ArbitrumGoerli.USDC
+  }[sourceChainId]
+}
+function parseTransferToMergedTransaction(
+  transfer: PendingCCTPTransfer | CompletedCCTPTransfer,
+  chainId: ChainId,
+  isPending: boolean
+): MergedTransaction {
+  const depositStatus = isPending
+    ? DepositStatus.CCTP_SOURCE_SUCCESS
+    : DepositStatus.CCTP_DESTINATION_SUCCESS
+
+  const { messageSent } = transfer
+  let status = 'Unconfirmed'
+  let resolvedAt = null
+  let receiveMessageTransactionHash = null
+
+  if ('messageReceived' in transfer) {
+    const { messageReceived } = transfer
+    status = 'Executed'
+    resolvedAt = getStandardizedTimestamp(
+      (parseInt(messageReceived.blockTimestamp, 10) * 1_000).toString()
+    )
+    receiveMessageTransactionHash = messageReceived.transactionHash
+  }
+  const sourceChainId = getSourceChainIdFromSourceDomain(
+    parseInt(messageSent.sourceDomain, 10),
+    chainId
+  )
+  const isDeposit =
+    parseInt(messageSent.sourceDomain, 10) === ChainDomain.Mainnet
+
+  return {
+    sender: messageSent.sender,
+    destination: messageSent.recipient,
+    direction: isDeposit ? 'deposit' : 'withdraw',
+    status,
+    createdAt: getStandardizedTimestamp(
+      (parseInt(messageSent.blockTimestamp, 10) * 1_000).toString()
+    ),
+    resolvedAt,
+    txId: messageSent.transactionHash,
+    asset: 'USDC',
```

Comment:
Should it be different based on deposit/withdrawal?

Reply:
I don't think so, in both case, CCTP transfers asset would native 'USDC'